packages/core/src/components/Form.js

```diff
@@ -31,134 +36,39 @@ export default class Form extends Component {
+    this.state = getStateFromProps(props, props.formData);
+  }
+
+  formElement = null;
+
+  componentDidMount() {
     if (
       this.props.onChange &&
       !deepEquals(this.state.formData, this.props.formData)
     ) {
-      this.props.onChange(this.state);
+      handleChange(this.props, this.state);
     }
-    this.formElement = null;
   }
-
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const nextState = this.getStateFromProps(nextProps, nextProps.formData);
+  componentDidUpdate(prevProps) {
     if (
-      !deepEquals(nextState.formData, nextProps.formData) &&
-      !deepEquals(nextState.formData, this.state.formData) &&
-      this.props.onChange
+      this.props.onChange &&
+      !deepEquals(this.state.formData, this.props.formData)
     ) {
-      this.props.onChange(nextState);
+      handleChange(this.props, this.state);
     }
-    this.setState(nextState);
   }
 
-  getStateFromProps(props, inputFormData) {
-    const state = this.state || {};
-    const schema = "schema" in props ? props.schema : this.props.schema;
-    const uiSchema = "uiSchema" in props ? props.uiSchema : this.props.uiSchema;
-    const edit = typeof inputFormData !== "undefined";
-    const liveValidate =
-      "liveValidate" in props ? props.liveValidate : this.props.liveValidate;
-    const mustValidate = edit && !props.noValidate && liveValidate;
-    const rootSchema = schema;
-    const formData = getDefaultFormState(schema, inputFormData, rootSchema);
-    const retrievedSchema = retrieveSchema(schema, rootSchema, formData);
-    const customFormats = props.customFormats;
-    const additionalMetaSchemas = props.additionalMetaSchemas;
-
-    const getCurrentErrors = () => {
-      if (props.noValidate) {
-        return { errors: [], errorSchema: {} };
-      } else if (!props.liveValidate) {
-        return {
-          errors: state.schemaValidationErrors || [],
-          errorSchema: state.schemaValidationErrorSchema || {},
-        };
-      }
-      return {
-        errors: state.errors || [],
-        errorSchema: state.errorSchema || {},
-      };
-    };
-
-    let errors,
-      errorSchema,
-      schemaValidationErrors,
-      schemaValidationErrorSchema;
-    if (mustValidate) {
-      const schemaValidation = this.validate(
-        formData,
-        schema,
-        additionalMetaSchemas,
-        customFormats
-      );
-      errors = schemaValidation.errors;
-      errorSchema = schemaValidation.errorSchema;
-      schemaValidationErrors = errors;
-      schemaValidationErrorSchema = errorSchema;
-    } else {
-      const currentErrors = getCurrentErrors();
-      errors = currentErrors.errors;
-      errorSchema = currentErrors.errorSchema;
-      schemaValidationErrors = state.schemaValidationErrors;
-      schemaValidationErrorSchema = state.schemaValidationErrorSchema;
+  static getDerivedStateFromProps(props, state) {
+    if (!deepEquals(state.lastProps, props)) {
```

Comment:
Why the reliance on deep equals for comparing property references? Does RJSF really want to support mutating object references? Such an action will not cause react to re-render the component anyway, so supporting this seems to be just a big performance sink.

We are working with very large forms (several hundered fields), and this area has been a pain point for us. I have an outstanding question about the issue here

Reply:
Hi @RoboPhred, from my view there is really no reason to use the deepEquals. I didn't spend any thought in reevaluating this, since I just wanted to have a PR with no breaking changes. I tried to replace deepEquals with shallow equals, and it didn't seem to break any tests, so I just pushed this change. We'll see if it passes review. :)

Reply:
Whoops, that change might not work, as you will always receive a new core props reference each time react re-renders, meaning it will try to make a new state every render.  You should compare each individual property to its previous value.  

You might be able to get away with something like:

```js
const keys = Object.keys(props);
if (keys.some(key => state.lastProps[key] != props[key]) {
  // regenerate state
}
```

I am a bit worried about adding or removing properties, but I /think/ that if you have an undefined property passed in react, react will actively create that property and set its value as undefined, meaning the property will still show up in Object.keys().  I don't think there are any cases where we might have a property defined in lastProps but not defined in props.

For reference, you can see my previous attempt at this [here](https://github.com/rjsf-team/react-jsonschema-form/pull/1634).  In that case, I individually checked the props by name, but I think we can get away with a bulk scan over the new properties.

Reply:
Ah, good catch. I added an additional check that checked the length of the number of properties, which I think will fix the case of any removed properties not triggering an onChange event. Not sure why the Netlify CI tests are breaking, but I think that's unrelated.